packages/calling/calling/src/CallingClient/line/index.ts

```diff
@@ -0,0 +1,238 @@
+    phoneNumber?: string,
+    extension?: string,
+    voicemail?: string
+  ) {
+    super();
+    this.lineId = uuid();
+    this.userId = userId;
+    this.clientDeviceUri = clientDeviceUri;
+    this.status = status;
+    this.phoneNumber = phoneNumber;
+    this.extension = extension;
+    this.voicemail = voicemail;
+
+    this.#sdkConnector = SDKConnector;
+    this.#webex = this.#sdkConnector.getWebex();
+    this.#mutex = mutex;
+
+    this.#primaryMobiusUris = primaryMobiusUris;
+    this.#backupMobiusUris = backupMobiusUris;
+
+    this.#sdkConfig = config;
+    const serviceData = this.#sdkConfig?.serviceData?.indicator
+      ? this.#sdkConfig.serviceData
+      : {indicator: ServiceIndicator.CALLING, domain: ''};
+
+    const logLevel = this.#sdkConfig?.logger?.level ? this.#sdkConfig.logger.level : LOGGER.ERROR;
+    validateServiceData(serviceData);
+
+    this.registration = createRegistration(
+      this.#webex,
+      serviceData,
+      this.#mutex,
+      this.lineEmitter,
+      callingClientEmitter,
+      logLevel
+    );
+
+    this.registration.setStatus(MobiusStatus.DEFAULT);
+    log.setLogger(logLevel, LINE_FILE);
+  }
+
+  /**
+   * Wrapper to for device registration.
+   *
+   * @param retry - Retry in case of errors.
+   */
+  public async register() {
+    await this.#mutex.runExclusive(async () => {
+      this.registration.setStatus(MobiusStatus.DEFAULT);
+      this.emit(LINE_EVENTS.CONNECTING);
+
+      this.registration.setMobiusServers(this.#primaryMobiusUris, this.#backupMobiusUris);
+      await this.registration.triggerRegistration();
+    });
+  }
+
+  /**
+   * Wrapper to for device  deregister.
+   */
+  public async deregister() {
+    this.registration.deregister();
+  }
+
+  /**
+   * Request for the Device status.
+   *
+   * @param deviceInfo - Registered device Object.
+   * @returns Promise.
+   */
+  public async sendKeepAlive(deviceInfo: IDeviceInfo): Promise<void> {
+    const deviceId = deviceInfo.device?.deviceId as string;
+    const interval = deviceInfo.keepaliveInterval as number;
+    const url = deviceInfo.device?.uri as string;
+
+    const logContext = {
+      file: LINE_FILE,
+      method: KEEPALIVE_UTIL,
+    };
+
+    this.registration.clearKeepaliveTimer();
+
+    log.info(
+      `Fetched details :-> deviceId: ${deviceId}, interval :-> ${interval}, url: ${url}`,
+      logContext
+    );
+
+    if (this.registration.isDeviceRegistered()) {
+      this.registration.startKeepaliveTimer(url, interval);
+    } else {
+      log.warn('Device is not active, exiting.', logContext);
+    }
+  }
+
+  private fillLine(deviceInfo: IDeviceInfo) {
+    const {
+      device,
+      keepaliveInterval,
+      callKeepaliveInterval,
+      rehomingIntervalMin,
+      rehomingIntervalMax,
+      voicePortalNumber,
+      voicePortalExtension,
+    } = deviceInfo;
+
+    this.mobiusDeviceId = device?.deviceId;
```

Comment:
Is the device object optional inside deviceInfo? Please check

Reply:
Yes device is optional inside IDeviceInfo. Do we want to make it mandatory?

Reply:
No this is fine. I just wanted to check